File: valid_parentheses/valid_parentheses.py
def isValid(self, s: str) -> bool:
    tracker = []
    parentheses = {")": "(", "}": "{", "]": "["}
    current = None
    for letter in s:
        if letter in parentheses.values():
            tracker.append(letter)
            current = letter
        elif letter in parentheses.keys():
            if current == parentheses[letter]:
                tracker.pop()
                if len(tracker) == 0:
                    current = None
                else: 
                    current = tracker[len(tracker) - 1]
            else:
                return False
    if len(tracker) == 0:
        return True
    else:
        return False

File: valid_parentheses/Valid_parentheses.java
package valid_parentheses;

import java.util.HashMap;
import java.util.Map;
import java.util.Stack;

public class Valid_parentheses {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        Map<Character, Character> parentheses = new HashMap<>();
        parentheses.put(')', '(');
        parentheses.put('}', '{');
        parentheses.put(']', '[');
        for (int i = 0; i < s.length(); i++) {
            char letter = s.charAt(i);
            if (parentheses.containsKey(letter)) {
                char topElement = stack.empty() ? '#' : stack.pop();
                if (topElement != parentheses.get(letter)) {
                    return false;
                }
            } else {
                stack.push(letter);
            }
        }
        return stack.isEmpty();
    }

    public void test() {
        System.out.print(this.isValid("()"));
    }
}
